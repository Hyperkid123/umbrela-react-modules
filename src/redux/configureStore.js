import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
