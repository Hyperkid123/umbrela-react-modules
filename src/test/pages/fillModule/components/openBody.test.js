import React from 'react'
import OpenBody, {OpenBody as Snapshot} from '../../../../pages/fillModule/components/openBody';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Open body component', () => {
  const initialProps = {};
  beforeEach(() => {
    initialProps.answer = {};
    initialProps.answerOpenQuestion = jest.fn();
    initialProps.translate = jest.fn();
    initialProps.questionId = 0;
    initialProps.answer = '';
  });

  it('should render correctly empty', () => {
    const tree = shallow(
      <Snapshot {...initialProps}/>
    )
    expect(toJson(tree)).toMatchSnapshot();
  })

  const mockStore = configureStore();
  let initialState = {};
  let store = {};
  beforeEach(() => {
    initialState = {
      answers: {},
      locale: {
        languages: [{
          name: "English",
          code: 'en',
          active: true,
        }],
        translations: {}
      }
    }
  });

  it('should create component correctly', () => {
    store = mockStore(initialState);
    let wrapper = shallow(
      <OpenBody questionId={0} store={store}/>);
    expect(wrapper).toBeDefined();
  });

  it('should return correct answer value', () => {
    const wrapper = shallow(
      <Snapshot {...initialProps}/>
    )
    expect(wrapper.instance().getOpenValue()).toEqual('');
    wrapper.setProps({answer: 'foo'});
    expect(wrapper.instance().getOpenValue()).toEqual('foo');
  })
});
