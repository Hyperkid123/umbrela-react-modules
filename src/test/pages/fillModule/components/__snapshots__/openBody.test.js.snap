// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Open body component should render correctly empty 1`] = `
<styled.div
  column={true}
  grow={true}
>
  <TextField
    fullWidth={true}
    key="openOption_0"
    multiline={true}
    onChange={[Function]}
    required={false}
    select={false}
    type="text"
    value=""
  />
  <TextFieldComent
    alignRight={true}
    error={false}
    label="0 undefined 1500 undefined"
  />
</styled.div>
`;
